catroid/src/main/java/org/catrobat/catroid/formulaeditor/function/BinaryFunction.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.formulaeditor.function
+
+class BinaryFunction(private val action: BinaryFunctionAction) : FormulaFunction, BinaryFunctionAction {
+    override fun execute(firstArgument: Double?, secondArgument: Double?): Any? {
+        if (firstArgument == null || secondArgument == null) {
+            return 0.0
+        }
+        return action.execute(firstArgument, secondArgument)
+    }
+
+    override fun execute(vararg args: Double?): Any? {
```

Comment:
Is there a case where null should be returned?
Maybe you find a way to make it more beautiful :)
I think we can simplify the logic :) (one of the executes might not be needed)
Also, have a look at the interface and its implementations maybe we can get rid of some duplicated logic ;)
If possible, use https://play.kotlinlang.org/byExample/05_Collections/05_existential for the nullcheck(s) please ;)